packages/contracts/contracts/factory/DAOFactory.sol

```diff
@@ -2,154 +2,143 @@
-    /// @param _mintConfig The configuration used to mint the newly created tokens.
-    /// @param _trustedForwarder The address of the trusted forwarder required for meta transactions.
-    function createERC20VotingDAO(
-        DAOConfig calldata _daoConfig,
-        VoteConfig calldata _voteConfig,
-        TokenFactory.TokenConfig calldata _tokenConfig,
-        TokenFactory.MintConfig calldata _mintConfig,
-        address _trustedForwarder
-    )
+    /// @notice Creates a new DAO and setup a number of plugins.
+    /// @param _daoSettings The DAO settings containing `trustedForwarder`, `name` and `metadata`.
+    /// @param _pluginSettings The list of plugin settings that will be installed after the DAO creation, containing `pluginSetup`, `pluginSetupRepo`, and `data`.
+    function createDao(DAOSettings calldata _daoSettings, PluginSettings[] calldata _pluginSettings)
         external
-        returns (
-            DAO dao,
-            ERC20Voting voting,
-            ERC20VotesUpgradeable token,
-            MerkleMinter minter
-        )
+        returns (DAO dao)
     {
-        if (_mintConfig.receivers.length != _mintConfig.amounts.length)
-            revert MintArrayLengthMismatch({
-                receiversArrayLength: _mintConfig.receivers.length,
-                amountsArrayLength: _mintConfig.amounts.length
-            });
-
-        dao = createDAO(_daoConfig, _trustedForwarder);
-
-        // Create token and merkle minter
-        dao.grant(address(dao), address(tokenFactory), dao.ROOT_PERMISSION_ID());
-        (token, minter) = tokenFactory.createToken(dao, _tokenConfig, _mintConfig);
-        dao.revoke(address(dao), address(tokenFactory), dao.ROOT_PERMISSION_ID());
-
-        daoRegistry.register(dao, msg.sender, _daoConfig.name);
-
-        voting = createERC20Voting(dao, token, _voteConfig);
-
-        setDAOPermissions(dao, address(voting));
+        // Check if no plugin is provided.
+        if (_pluginSettings.length == 0) {
+            revert NoPluginProvided();
+        }
 
-        emit DAOCreated(_daoConfig.name, address(token), address(voting));
-    }
-
-    /// @notice Creates a new DAO with the `AllowlistVoting` component installed.
-    /// @param _daoConfig The name and metadata hash of the DAO.
-    /// @param _voteConfig The configuration used to set up the the majority voting.
-    /// @param _allowlistVoters An array of addresses that are allowed to vote.
-    /// @param _trustedForwarder The address of the trusted forwarder required for meta transactions.
-    function createAllowlistVotingDAO(
-        DAOConfig calldata _daoConfig,
-        VoteConfig calldata _voteConfig,
-        address[] calldata _allowlistVoters,
-        address _trustedForwarder
-    ) external returns (DAO dao, AllowlistVoting voting) {
-        dao = createDAO(_daoConfig, _trustedForwarder);
+        // Create DAO.
+        dao = _createDAO(_daoSettings);
```

Comment:
I would name it to `createdDao` or `daoInCreation` to discriminate it from a `managingDao` (see https://github.com/aragon/core/pull/113/files#r995385900 where this was misinterpreted).

Reply:
what about `newDao` ? because `daoInCreation` sounds like a ongoing progress which is not exactly, dao is created already.